quic/s2n-quic-core/src/transport/error.rs

```diff
@@ -165,6 +166,72 @@ macro_rules! impl_errors {
             }
         }
 
+        impl aggregate::AsVariant for Code {
+            const VARIANTS: &'static [aggregate::info::Variant] = &{
+                use aggregate::info::{Variant, Str};
+
+                const fn count(_v: u64) -> usize {
+                    1
+                }
+
+                const QUIC_VARIANTS: usize = 0 $( + count($code))*;
+
+                const TLS: &'static [Variant] = tls::Error::VARIANTS;
+
+                let mut array = [
+                    Variant { name: Str::new("\0"), id: 0 };
+                    QUIC_VARIANTS + TLS.len() + 1
+                ];
+
+                let mut idx = 0;
+
+                $(
+                    array[idx] = Variant {
+                        name: Str::new(concat!("QUIC_", stringify!($name), "\0")),
+                        id: $code,
+                    };
+                    idx += 1;
+                )*
+
+                let mut tls_idx = 0;
+                while tls_idx < TLS.len() {
+                    let variant = TLS[tls_idx];
+                    array[idx] = Variant {
+                        name: variant.name,
+                        id: variant.id | 0x100,
+                    };
+                    idx += 1;
+                    tls_idx += 1;
+                }
+
+                array[idx] = Variant {
+                    name: Str::new("QUIC_UNKNOWN_ERROR\0"),
+                    id: u32::MAX as _,
+                };
+
+                array
+            };
+
+            #[inline]
+            fn variant_idx(&self) -> usize {
+                let mut idx = 0;
+                let code = self.0.as_u64();
+
+                $(
+                    if code == $code {
+                        return idx;
+                    }
+                    idx += 1;
+                )*
+
+                if (0x100..=0x1ff).contains(&code) {
```

Comment:
Similarly this seems like it ought to come with an assert of some kind?

Reply:
Not sure I follow _what_ you're wanting to assert here?

Reply:
Might be moot now, but that 0x1ff actually bounds the upper range of these codes -- and that 0x100 is the same across both places we put the constant, but that's a smaller point :)